src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageSearch/PackageSearchRunner.cs

```diff
@@ -0,0 +1,201 @@
+                int completedTaskIndex = searchRequests.FindIndex(t => t.Item1 == completedTask); ;
+                PackageSource source = searchRequests[completedTaskIndex].Item2;
+
+                await packageSearchResultRenderer.Add(source, completedTask);
+
+                searchRequests.RemoveAt(completedTaskIndex);
+            }
+
+            packageSearchResultRenderer.Finish();
+        }
+
+        /// <summary>
+        /// Builds a task that performs package search operation.
+        /// </summary>
+        /// <param name="source">A package source/endpoint.</param>
+        /// <param name="packageSearchArgs">Package search arguments</param>
+        /// <param name="cancellationToken">A token to monitor for cancellation requests.</param>
+        /// <returns>A list of tasks that perform package search operations.</returns>
+        private static Task<IEnumerable<IPackageSearchMetadata>> SearchAsync(
+            PackageSource source,
+            PackageSearchArgs packageSearchArgs,
+            CancellationToken cancellationToken)
+        {
+            return Task.Run(async () =>
+            {
+                var repository = Repository.Factory.GetCoreV3(source);
+                var resource = await repository.GetResourceAsync<PackageSearchResource>(cancellationToken);
+
+                if (resource == null)
+                {
+                    return null;
+                }
+
+                return await resource.SearchAsync(
+                    packageSearchArgs.SearchTerm,
+                    new SearchFilter(includePrerelease: packageSearchArgs.Prerelease),
+                    packageSearchArgs.Skip,
+                    packageSearchArgs.Take,
+                    packageSearchArgs.Logger,
+                    cancellationToken);
+            });
+        }
+
+        /// <summary>
+        /// Builds a task that perform exact match package search operation.
+        /// </summary>
+        /// <param name="source">A package source.</param>
+        /// <param name="packageSearchArgs">Package search arguments</param>
+        /// <param name="cancellationToken">A token to monitor for cancellation requests.</param>
+        /// <returns>A list of tasks that perform exact match package search operations.</returns>
+        private static Task<IEnumerable<IPackageSearchMetadata>> GetPackageAsync(
+            PackageSource source,
+            PackageSearchArgs packageSearchArgs,
+            CancellationToken cancellationToken)
+        {
+            return Task.Run(async () =>
+            {
+                using var cache = new SourceCacheContext();
+                var repository = Repository.Factory.GetCoreV3(source);
+                var resource = await repository.GetResourceAsync<PackageMetadataResource>();
+
+                if (resource == null)
+                {
+                    return null;
+                }
+
+                return await resource.GetMetadataAsync(
```

Comment:
looks like Take and Skip are ignored in this codepath (the exact match one). should those switches error if they are passed along with --exact-match?

Reply:
Yes that is correct. `--exact-match` provides with only 1 or 0 search results, as a result, there is no need to use skip or take options. Printing an error was something I thought about, but I am not sure if we should interrupt users for using them, as they are harmless. But I believe, mentioning that they are ignored during `--exact-match` in the help output would bring more clarity to users.

Reply:
I have updated the help output to mention this.